front/src/modules/timesStops/hooks/useOutputTableData.ts

```diff
@@ -4,154 +4,124 @@ import { keyBy } from 'lodash';
-      ...pathStep,
-      positionOnPath: path.path_item_positions[index],
-    }));
-  }, [path, selectedTrainSchedule]);
-
-  const pathStepsWithOpPointIndices = useMemo(
-    () =>
-      selectedTrainSchedule.path
-        .filter((pathStep): pathStep is TrainScheduleBasePathWithUic => 'uic' in pathStep)
-        .map((pathStep) => {
-          const correspondingOpPointIndex = suggestedOperationalPoints.findIndex(
-            (sugOpPoint) =>
-              'uic' in pathStep &&
-              sugOpPoint.uic === pathStep.uic &&
-              sugOpPoint.ch === pathStep.secondary_code
-          );
-          return {
-            ...pathStep,
-            correspondingOpPointIndex,
-          };
-        }),
-    [selectedTrainSchedule, suggestedOperationalPoints]
-  );
-  const pathStepsByUic = keyBy(
-    pathStepsWithOpPointIndices,
-    (pathStep) => `${pathStep.uic}-${pathStep.secondary_code}`
-  );
-
-  const outputTableData = useMemo(() => {
-    const pathStepRows = pathStepsWithPositionOnPath.map((pathStep) => {
-      const schedule = scheduleByAt[pathStep.id];
-      const scheduleData = computeScheduleData(schedule);
-      return { ...pathStep, ...formatScheduleData(scheduleData) };
-    });
+  const scheduleByAt: Record<string, ScheduleEntry> = keyBy(selectedTrainSchedule?.schedule, 'at');
 
-    const suggestedOpRows = suggestedOperationalPoints.map((sugOpPoint, sugOpIndex) => {
-      const opPoint = operationalPoints.find((op) => op.id === sugOpPoint.opId);
-      if (!opPoint) {
-        return sugOpPoint;
-      }
-      const nextOpPoint = findNextScheduledOpPoint(
-        operationalPoints,
-        pathStepsWithOpPointIndices,
-        sugOpIndex
+  const pathStepRows = useMemo(() => {
+    if (!path || !selectedTrainSchedule || !trainSummary?.pathItemTimes) return [];
+
+    const startDatetime = new Date(selectedTrainSchedule.start_time);
+
+    const result: (TimeStopsRow & { positionOnPath: number })[] = [];
+    let previousTime = new Date(selectedTrainSchedule.start_time);
+
+    for (let index = 0; index < selectedTrainSchedule.path.length; index += 1) {
+      const pathStep = selectedTrainSchedule.path[index];
+      const schedule: ScheduleEntry | undefined = scheduleByAt[pathStep.id];
+
+      const computedArrival = new Date(
+        trainSummary.pathItemTimes.final[index] + startDatetime.getTime()
+      );
+
+      const { stopFor, shortSlipDistance, onStopSignal, calculatedDeparture } = formatSchedule(
+        computedArrival,
+        schedule
+      );
+      const { theoreticalMargin, theoreticalMarginSeconds, calculatedMargin, diffMargins } =
+        computeMargins(selectedTrainSchedule, index, trainSummary.pathItemTimes);
+
+      const inputs = computeInputDatetimes(startDatetime, previousTime, schedule, {
+        isDeparture: index === 0,
+      });
+      const { arrival, departure } = inputs;
+      previousTime = inputs.prev;
```

Comment:
This property could be renamed `refDate` (just a suggestion) or something like that (`previousTime` is actually a `Date`)

```suggestion
      lastRefDate = inputs.refDate;
```

Reply:
You're right, done !